pkg/distributor/distributor.go

```diff
@@ -598,9 +599,12 @@ func (d *Distributor) Push(ctx context.Context, req *cortexpb.WriteRequest) (*co
 	validatedSamples := 0
 	validatedExemplars := 0
 
-	if d.limits.AcceptHASamples(userID) && len(req.Timeseries) > 0 {
-		cluster, replica := findHALabels(d.limits.HAReplicaLabel(userID), d.limits.HAClusterLabel(userID), req.Timeseries[0].Labels)
-		removeReplica, err = d.checkSample(ctx, userID, cluster, replica)
+	// Cache user limit with overrides.
+	limits, _ := validation.NewOverrides(*(d.limits.GetOverridesForUser(userID)), nil)
```

Comment:
Do we need to create the `validation.NewOverrides` object? We simply use the limit returned by `d.limits.GetOverridesForUser(userID)`?

Reply:
The limit struct doesn't  implement some interface, if change it, we can get the value by exported field rather than method.